plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1204,7 +1206,7 @@ func (c *PodAffinityChecker) satisfiesPodsAffinityAntiAffinity(pod *v1.Pod, node
 		if err != nil {
 			glog.Errorf("Cannot schedule pod %+v onto node %v, because of PodAffinityTerm %v, err: %v",
 				podName(pod), node.Name, term, err)
-			return false
+			return ErrPodAffinityRulesNotMatch, nil
```

Comment:
@guangxuli I think you are right that we should return appropriate error here and in the cases below. One thing to node though is that [`podFitsOnNode`](https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/core/generic_scheduler.go#L327) checks the error and accounts failing nodes differently than simple predicate failure when error is set. In current scheduler code, we don't return error for these case, so `podFitsOnNode` considers the errors as predicate failures. After this PR, the behavior will be a bit different. In my opinion this change should be fine and will make the scheduler code and its stats more accurate. However, we should make sure that we don't break any test or assumptions.

Reply:
agree, after this PR we can give the specific predicate failure reasons and error details.

Reply:
yes, i will make a big effort to reduce the modification of original test cases.